src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java

```diff
@@ -0,0 +1,130 @@
+import com.google.cloud.tools.app.api.deploy.AppEngineFlexibleStaging;
+import com.google.cloud.tools.app.api.deploy.AppEngineStandardStaging;
+import com.google.cloud.tools.app.api.devserver.AppEngineDevServer;
+import com.google.cloud.tools.app.impl.cloudsdk.CloudSdkAppEngineDeployment;
+import com.google.cloud.tools.app.impl.cloudsdk.CloudSdkAppEngineDevServer;
+import com.google.cloud.tools.app.impl.cloudsdk.CloudSdkAppEngineFlexibleStaging;
+import com.google.cloud.tools.app.impl.cloudsdk.CloudSdkAppEngineStandardStaging;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessOutputLineListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.sdk.CloudSdk;
+
+import org.apache.maven.plugin.logging.Log;
+
+/**
+ * Factory for App Engine dependencies.
+ */
+public class CloudSdkAppEngineFactory implements AppEngineFactory {
+
+  protected CloudSdkFactory cloudSdkFactory = new CloudSdkFactory();
+  protected CloudSdkMojo mojo;
+
+  public CloudSdkAppEngineFactory(CloudSdkMojo mojo) {
+    this.mojo = mojo;
+  }
+
+  @Override
+  public AppEngineStandardStaging standardStaging() {
+    return cloudSdkFactory.standardStaging(defaultCloudSdkBuilder().build());
+  }
+
+  @Override
+  public AppEngineFlexibleStaging flexibleStaging() {
+    return cloudSdkFactory.flexibleStaging();
+  }
+
+  @Override
+  public AppEngineDeployment deployment() {
+    return cloudSdkFactory.deployment(defaultCloudSdkBuilder().build());
+  }
+
+  @Override
+  public AppEngineDevServer devServerRunSync() {
+    return new CloudSdkAppEngineDevServer(defaultCloudSdkBuilder().build());
+  }
+
+  @Override
+  public AppEngineDevServer devServerRunAsync(int startSuccessTimeout) {
+    CloudSdk.Builder builder = defaultCloudSdkBuilder()
+        .async(true)
+        .runDevAppServerWait(startSuccessTimeout);
+    return cloudSdkFactory.devServer(builder.build());
+  }
+
+  @Override
+  public AppEngineDevServer devServerStop() {
+    return cloudSdkFactory.devServer(null);
+  }
+
+  protected CloudSdk.Builder defaultCloudSdkBuilder() {
+
+    ProcessOutputLineListener lineListener = new DefaultProcessOutputLineListener(mojo.getLog());
+
+    return cloudSdkFactory.cloudSdkBuilder()
+        .sdkPath(mojo.cloudSdkPath)
+        .addStdOutLineListener(lineListener)
+        .addStdErrLineListener(lineListener)
+        .appCommandMetricsEnvironment(mojo.pluginDescriptor.getArtifactId())
+        .appCommandMetricsEnvironmentVersion(mojo.pluginDescriptor.getVersion());
+  }
+
+  /**
+   * Default output listener that copies output to the Maven Mojo logger with a 'GCLOUD: ' prefix.
+   */
+  protected static class DefaultProcessOutputLineListener implements ProcessOutputLineListener {
+
+    private Log log;
+
+    public DefaultProcessOutputLineListener(Log log) {
+      this.log = log;
+    }
+
+    @Override
+    public void outputLine(String line) {
+      log.info("GCLOUD: " + line);
+    }
+  }
+
+  protected static class CloudSdkFactory {
```

Comment:
I feel like this class is not necessary. Does this help with testing somehow or something that isn't apparent yet?